### PR TITLE
fix(server): fix past key values logic

### PR DESCRIPTION
**Pull request by OlivierDehaene**
@njhill fyi

### CODE REVIEW

File: server/text_generation_server/models/flash_causal_lm.py [hunk 1]
```diff
@@ -378,7 +399,9 @@ def generate_token(
     ) -> Tuple[List[Generation], Optional[FlashCausalLMBatch]]:
         # Shortcut when batch_size == 1
         if len(batch) == 1:
-            # No need to slice this down
+            input_ids = batch.input_ids[0].view(-1)
+            # Slice to remove extra padding
+            # past_key_values = batch.past_key_values[:, :batch.input_lengths[0]] if batch.past_key_values is not None else None
```
Comment (njhill) [inline server/text_generation_server/models/flash_causal_lm.py:404]: The modeling code just uses offsets so no need to slice...

**Review by njhill — COMMENTED** [review]
Thanks @OlivierDehaene!

File: server/text_generation_server/models/flash_causal_lm.py [hunk 2]
```diff
@@ -405,21 +438,28 @@ def generate_token(
...
+                )
             # Set in batch in case it needs to be used later in concatenate()
             batch.past_pad = self.past_pad
             if len(batch) == 1:
                 # Preallocate tensor for bs = 1 case
                 batch.past_key_values = torch.nn.functional.pad(
-                    present, (0, 0, 0, 0, 0, 0, 0, batch.stopping_criterias[0].max_new_tokens)
+                    present,
+                    (0, 0, 0, 0, 0, 0, 0, batch.stopping_criterias[0].max_new_tokens),
                 )
```
[the remark this answers is not in the document]
Comment (OlivierDehaene) [reply server/text_generation_server/models/flash_causal_lm.py:458]: Yes I forgot to cleanup...

File: server/text_generation_server/models/flash_causal_lm.py (hunk 1 above)
Comment (OlivierDehaene) [reply server/text_generation_server/models/flash_causal_lm.py:404]: Yes I forgot to cleanup again...

File: server/text_generation_server/models/flash_causal_lm.py [hunk 3]
```diff
@@ -207,7 +210,17 @@ def filter(self, requests: List[generate_pb2.Request]) -> "FlashCausalLMBatch":
...
                 self.past_key_values[0],
-                (0, 0, 0, 0, 0, 0, 0, stopping_criterias[0].max_new_tokens - stopping_criterias[0].current_tokens)
+                (
+                    0,
+                    0,
+                    0,
+                    0,
+                    0,
+                    0,
+                    0,
```
[the remark this answers is not in the document]
Comment (OlivierDehaene) [reply server/text_generation_server/models/flash_causal_lm.py:220]: black doing black things =P

**Pull request merged**